File: Android/TestSuperModel/app/src/main/java/com/soonzik/testsupermodel/Tweet.java
package com.soonzik.testsupermodel;

import org.json.JSONObject;

/**
 * Created by kevin_000 on 21/03/2015.
 */
public class Tweet extends ActiveRecord {
    private int id = -1;
    private String msg = "";
    private User writer = null;

    public Tweet() {}

    public Tweet(JSONObject json) {
        super.createInstance(this, json, this.getClass());
    }

    @Override
    public String toString() {
        return (
                "id = " + Integer.toString(id)
                + " : msg = " + msg
                + " : writer = { " + (writer != null ? writer.toString() : "") + "  }"
                );
    }
}

File: iOS SoonZik/SoonZik/Models/Tweet.m
//
//  Tweet.m
//  SoonZik
//
//  Created by Maxime Sauvage on 25/08/2015.
//  Copyright (c) 2015 SoonZik - Maxime SAUVAGE. All rights reserved.
//

#import "Tweet.h"

@implementation Tweet

- (id)initWithJsonObject:(NSDictionary *)json {
    self = [super init];
    
    self.identifier = [[json objectForKey:@"id"] intValue];
    self.msg = [json objectForKey:@"msg"];
    NSDateFormatter *dateformat = [[NSDateFormatter alloc] init];
    [dateformat setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
    dateformat.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:1];
    self.date = [dateformat dateFromString:[json objectForKey:@"created_at"]];
    self.user = [[User alloc] initWithJsonObject:[json objectForKey:@"user"]];
    
    return self;
}

@end
